### DISCUSSION TITLE
Coral PCIe TPU in Proxmox via VM ( Debian ) without HW Accel?

### DISCUSSION THREAD

**Discussion by Vendo232** [blakeblackshear/frigate#6403] [Ask A Question]
Using EliteDesk HP G2 i5 6600T with Coral M.2 on Proxmox and VM Debian 11 . Was able to passthough the PCI / IOMMU card to the VM , install drivers and all works but I had to disable Hardware accelleration. Having the HW Accell enabled was giving me all sorts of issues and could not start any of my sreams.

I feel like I`m missing something but my CPU runs low and inference is at 6ms. All looks normal

Is it OK to run Frigate like this?
![image](https://user-images.githubusercontent.com/44098230/236579605-bd5b1de4-3ba0-4a7e-97e7-9770b88e9f1d.png)

![image](https://user-images.githubusercontent.com/44098230/236579307-aff28c43-e118-4c3f-bb15-4db1cc61e223.png)

**Comment by Vendo232** [comment]
here are the steps I used to add Coral to the VM

`nano /etc/default/grub`

for intel:

add

```
GRUB_CMDLINE_LINUX_DEFAULT="quiet intel_iommu=on"

GRUB_CMDLINE_LINUX_DEFAULT="quiet intel_iommu=on iommu=pt"
```

save
```
update-grub

nano /etc/modules
```

ad modules

```
vfio

vfio_iommu_type1

vfio_pci

vfio_virqfd

reboot
```


in VM

install coral drivers 2a ( 1-6 )

the add user to plugdev

`sudo usermod -aG plugdev $USER`

**Comment by NickM-27** [comment]
> Is it OK to run Frigate like this?

Sure, you are just going to have higher CPU usage than you could have
